### PR TITLE
Prevent the use of external facing API commands in backend consistently

### PR DESCRIPTION
**Pull request by okurz**
Some backend code (mis-)used the testapi "get_var" and other functions when
we should use the bmwqemu variable directly. This also fixes the check for
documentation of backend variables which relies on the variable use.

Used the following command for automatic replacement and some manual corrections:

```
sed -i -e "s/get_required_var([\"\']\([A-Za-z0-9_\$]*\)[\"\'])/\$bmwqemu::vars{\1} or die 'Need variable \1'/g" -e "s/get_var([\"\']\([A-Za-z0-9_\$]*\)[\"\'])/\$bmwqemu::vars{\1}/g" -e "s/get_var([\"\']\([A-Za-z0-9_\$]*\)[\"\'], [\"\']\([A-Za-z_0-9]*\)[\"\'])/\$bmwqemu::vars{\1} \/\/ \'\2\'/g" -e "s/check_var([\"\']\([A-Za-z0-9_\$]*\)[\"\'], [\"\']\([A-Za-z_0-9]*\)[\"\'])/\$bmwqemu::vars{\1} eq \'\2\'/g" $(git grep -l '_var' backend/)
```

### CODE REVIEW

**Review by kalikiana — APPROVED** [review]

File: doc/backend_vars.asciidoc [hunk 1]
```diff
@@ -43,6 +43,16 @@ UPLOAD_INACTIVITY_TIMEOUT;integer;300;Specifies the inactivity timeout in second
 
 |====================
 
+.ZVM backend
+[grid="rows",format="csv"]
+[options="header",cols="^m,^m,^m,v",separator=";"]
+|====================
+Variable;Values allowed;Default value;Explanation
+ZVM_HOST;string;;Sets the remote host name.
+ZVM_GUEST;string;;Sets the remote user name.
```
Comment (Martchus) [inline doc/backend_vars.asciidoc:52]: See comments in original PR.

File: doc/backend_vars.asciidoc [hunk 2]
```diff
@@ -72,6 +82,8 @@ IPMI_MC_RESET_TIMEOUT;integer;60;Counts to try to reach IPMI interface after mc
 IPMI_MC_RESET_PING_COUNT;integer;1;Ping counts that must be successful after mc reset
 IPMI_MC_RESET_IPMI_TRIES;integer;3;Maximum number of IPMI command tries that are conducted after mc reset
 IPMI_SOL_PERSISTENT_CONSOLE;boolean;1;Make SOL console persistent and don't reset it, enabled by default
+IPMI_$_;;;Internal iterator variable
+WORKER_HOSTNAME;string;undef;Worker host name
```
Comment (Martchus) [inline doc/backend_vars.asciidoc:86]:
```suggestion
WORKER_HOSTNAME;string;undef;Worker hostname
```

File: doc/backend_vars.asciidoc [hunk 3]
```diff
@@ -275,6 +295,11 @@ GENERAL_HW_POWEROFF_CMD;string;;Shell Command to power off the SUT (in CMD_DIR)
 GENERAL_HW_POWEROFF_ARGS;string;;Arguments to pass GENERAL_HW_POWEROFF_CMD Shell script
 GENERAL_HW_FLASH_CMD;string;;Shell Command to flash a disk image on SUT (in CMD_DIR), optional
 GENERAL_HW_FLASH_ARGS;string;;Arguments to pass GENERAL_HW_FLASH_CMD Shell script
+HDDSIZEGB;integer;10;Creates HDD with specified size in GiB
+HDD_$i;filename;;Filename of HDD image to be used for machine.
+HDDSIZEGB_$i;integer;;Creates HDD with specified size in GiB for corresponding HDD
+NUMDISKS;integer;1;Number of disks attached to machine
+WORKER_HOSTNAME;string;undef;Worker host name
```
Comment (Martchus) [inline doc/backend_vars.asciidoc:302]:
```suggestion
WORKER_HOSTNAME;string;undef;Worker hostname
```

File: doc/backend_vars.asciidoc (hunk 1 above)
Comment (okurz) [reply doc/backend_vars.asciidoc]: fixed

File: doc/backend_vars.asciidoc (hunk 2 above)
Comment (okurz) [reply doc/backend_vars.asciidoc]: done

File: doc/backend_vars.asciidoc (hunk 3 above)
Comment (okurz) [reply doc/backend_vars.asciidoc]: done

**Review by Martchus — APPROVED** [review]

**Comment by okurz** [comment]
codecov/patch not met. That was expected. Merging manually.

**Comment by jlausuch** [comment]
This breaks VMWare booting in all our tests. Is there a chance to revert it or should we wait for a fix?
See https://progress.opensuse.org/issues/104992

**Comment by Martchus** [comment]
@jlausuch I've been preparing a fix here: https://github.com/os-autoinst/os-autoinst/pull/1923

**Comment by jlausuch** [comment]
> @jlausuch I've been preparing a fix here: #1923

ok thanks!

**Pull request merged**